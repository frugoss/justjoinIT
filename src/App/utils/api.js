export let API_HOST = 'http://localhost:7000'
if (process.env.NODE_ENV === 'test') {
  API_HOST = 'http://localhost'
}
