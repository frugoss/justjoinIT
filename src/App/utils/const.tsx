

export interface userInterface {
    auth?: boolean,
    name?: string,
    userID?: string,
    loggedPopup?: boolean,
    offPopup?: boolean,
    createPopup?: boolean,
    addPopup?: boolean
}


export interface offerInterface {
    _id: string,
    agreements: string,
    city: string,
    company: string,
    companyIndustry: string,
    companySize: string,
    coordinates: [{ lat: number, lng: number }],
    currency: string,
    description: string,
    email: string,
    employment: string,
    experience: string,
    logo: string | undefined,
    maxSalary: number,
    minSalary: number,
    street: string,
    techStack: [{ language: string, lvl: number }],
    technology: [{ color: string, background: string, img: string, name: string }],
    title: string,
    userID: string,
    website: string,
}

export interface formInterface {
    _id?: string,
    agreements?: string,
    city?: string,
    company?: string,
    companyIndustry?: string,
    companySize?: string,
    coordinates?: [{ lat: number, lng: number }],
    currency?: string,
    description?: string,
    email?: string,
    employment?: string,
    experience?: string,
    logo?: string| undefined,
    maxSalary?: number,
    minSalary?: number,
    street?: string,
    techStack: { language?: string, lvl?: number }[],
    technology?: [{ color: string, background: string, img: string, name: string }],
    title?: string,
    userID: string,
    website?: string,
}


export interface applicationInterface {
    checkbox: boolean,
    email: string,
    file: string,
    message: string,
    name: string,
    offerID: string
}

export interface filtersInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: number,
    salarymin: number,
}

export interface matchInterface{
    city: string,
    language: string,
    experience: string,
    salarymax: string,
    salarymin: string,
}

export interface formValuesCV  {
    name:string,
    email:string,
    message: string,
    checkbox: string,
    file: File,
    offerID: string
}

export interface languageInterface {
    name: string,
    to: string,
    img: string,
    color: string,
    background: string
}



export interface mobileViewModeInterface {
    view: string,
    header: boolean
}

export interface cityInterface {
    name: string,
    to: string,
    coordinates: {lat:number, lng:number}
}
