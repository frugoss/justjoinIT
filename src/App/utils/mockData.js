export const offersList = [
  {
    "_id": "5e9cfa222f9c0e0965b86034",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;baC",
        "color": "rgb(255, 199, 6)",
        "name": "javascript",
        "background": "rgb(255, 199, 6), rgb(255, 175, 0)"
      }
    ],
    "techStack": [
      {
        "language": "JavaScript",
        "lvl": 5
      },
      {
        "language": "ES7",
        "lvl": 3
      }
    ],
    "logo": "data:image/png;basI=",
    "coordinates": [
      {
        "lat": 54.4399555,
        "lng": 18.5440814
      }
    ],
    "city": "Sopot",
    "street": "Tuwima 20",
    "agreements": "Informujemy, że administratorem danych jest dddd z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "company": "Travers",
    "companyType": "Software House",
    "companyIndustry": "Medicine",
    "currency": "PLN",
    "employment": "Mandate Contract",
    "experience": "Senior",
    "title": "Javascript developer",
    "minSalary": 15000,
    "maxSalary": 25000,
    "description": "<div><!--block--><strong>Who we are looking for<br></strong><br></div><ul><li><!--block--><strong>Senior Angular Developer (3+ experience)</strong></li><li><!--block-->Good analytical skills (problem solving)</li><li><!--block-->Good communication skills</li><li><!--block-->Enjoy helping other Engineers improve and become better</li><li><!--block-->Very good English speaking skills</li></ul><div><!--block--><strong><br>What we offer<br></strong><br></div><ul><li><!--block-->Working in a Development Team (4 team members) alongside backend team placed in London</li><li><!--block-->Lead a team of talented and collaborative engineers&nbsp;</li><li><!--block-->Help mentor and coach members of the team</li><li><!--block-->Help evolve development standards and practices</li><li><!--block-->Review Pull Requests from other developers</li><li><!--block-->Overall responsible for the technical quality of what the team is producing</li><li><!--block-->Collaborate with back-end developers, UX and UI designers to improve our products and features</li></ul><div><!--block--><strong><br>Benefits<br></strong><br></div><ul><li><!--block-->26 days of paid holidays per year (+ additionally paid bank holidays)</li><li><!--block-->30 days sick leave per year (paid 80%)</li><li><!--block--><strong>Flexible working hours</strong></li><li><!--block-->Work from home (1 day per week)</li><li><!--block--><strong>Taking part in development-related events</strong> (conferences, trainings)</li><li><!--block-->Private medical care</li><li><!--block-->MultiSport Plus Package</li></ul>",
    "companySize": "10 - 40",
    "website": "www.traversy.com"
  },
  {
    "_id": "5e9a411cfe04a70747f1a7d1",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(31, 94, 182)",
        "name": "python",
        "background": "rgb(31, 94, 182), rgb(31, 123, 196)"
      }
    ],
    "techStack": [
      {
        "language": "Javascript",
        "lvl": 3
      },
      {
        "language": "C#",
        "lvl": 2
      }
    ],
    "logo": "data:image/png;",
    "companyType": "Software House",
    "companySize": "10",
    "companyIndustry": "E-commerce",
    "experience": "Mid",
    "title": "Python Developer",
    "employment": "Permanent",
    "minSalary": 3000,
    "maxSalary": 10000,
    "currency": "EUR",
    "description": "<div><!--block-->Lorem ipsum dolor sit amet, consectetur adipiscing elit. In tincidunt maximus sem vitae cursus. Integer ullamcorper quam quis molestie dictum. Nam eros augue, aliquam vitae tristique in, laoreet ac massa. Suspendisse ultricies vulputate elit, scelerisque pellentesque turpis tempor in. Donec neque est, feugiat sit amet tempor ut, gravida sit amet ligula. Donec elit ante, pharetra ac ex a, malesuada molestie velit. Vivamus cursus risus mi, quis maximus justo convallis vitae.  Nullam ultrices porttitor quam, non cursus risus convallis vel. Mauris feugiat lectus eu arcu tempor, eget hendrerit eros imperdiet. Ut vitae sodales nisi. Nam eu dolor eu libero tincidunt maximus non vitae odio. Cras neque orci, dignissim eu feugiat ac, consectetur ut nisl. Maecenas euismod sem orci. Phasellus eget nulla et sapien vulputate mollis vel tempor sem. Nulla cursus pharetra lacus, lobortis porta nulla eleifend et. Curabitur quis arcu id velit gravida euismod vitae eu turpis. In suscipit, lectus vitae maximus iaculis, mauris massa porttitor est, vel porttitor orci massa ut quam. Morbi in sapien sit amet nisl faucibus posuere.  Quisque sit amet finibus leo. Aenean placerat sit amet nulla rutrum auctor. Curabitur scelerisque eleifend purus, quis bibendum lectus venenatis vitae. Quisque pellentesque lorem ex, eu facilisis magna convallis eget. Aliquam eget fermentum augue. Ut sollicitudin dui elit, eu consectetur mi rutrum in. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nam at massa interdum, feugiat augue ac, fringilla mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus ut velit vulputate nulla luctus aliquet.  Sed ornare viverra felis vitae tincidunt. Pellentesque hendrerit massa sit amet nibh viverra lacinia. Duis vitae metus vel libero tristique euismod. Sed congue consequat purus, quis convallis ex venenatis id. Donec lacus massa, tincidunt vel erat quis, accumsan euismod erat. Curabitur leo dolor, aliquet eget fringilla a, fermentum pulvinar nisl. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.  Proin eleifend sodales ligula in hendrerit. Nam volutpat, enim id varius bibendum, metus orci venenatis nibh, nec vestibulum ex dolor sit amet ipsum. Cras blandit enim vitae justo lobortis, vel efficitur sem luctus. Duis non dolor id erat consectetur eleifend. Nullam ultrices dictum lacus faucibus aliquet. Vestibulum eget diam cursus, fermentum turpis id, aliquet odio. Quisque vitae bibendum nibh, faucibus malesuada mi. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Mauris dictum risus vel metus convallis, vitae cursus metus faucibus. Cras placerat rhoncus enim non tempor. Integer augue odio, efficitur non diam et, luctus lobortis mi. Suspendisse laoreet vestibulum ex non pretium. Nullam rhoncus, risus nec placerat laoreet, dui neque ultrices metus, vitae congue ligula quam a magna. Nullam sodales lacus in aliquet sagittis.</div>",
    "city": "Gdańsk",
    "street": "Tuwima",
    "coordinates": [
      {
        "lat": 54.3659778,
        "lng": 18.6342856
      }
    ],
    "email": "1234@gmail.com",
    "agreements": "Informujemy, że administratorem danych jest _____xDD_________ z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją.",
    "website": "www.lion.com",
    "company": "Lion",
    "remote": true
  },
  {
    "_id": "5e9cf7062f9c0e0965b86029",
    "userID": "5e8cc6d853004b069507091d",
    "technology": [
      {
        "img": "data:image/png;base64",
        "color": "rgb(47, 207, 187)",
        "name": "c",
        "background": "rgb(47, 207, 187), rgb(55, 221, 156)"
      }
    ],
    "techStack": [
      {
        "language": "C",
        "lvl": 4
      },
      {
        "language": "Java",
        "lvl": 2
      },
      {
        "language": "Python",
        "lvl": 5
      }
    ],
    "logo": "data:image/jpeg;base64,",
    "currency": "USD",
    "companyType": "Startup",
    "companyIndustry": "Military",
    "email": "test",
    "employment": "B2B",
    "experience": "Mid",
    "city": "Gdynia",
    "coordinates": [
      {
        "lat": 54.51889,
        "lng": 18.53188
      }
    ],
    "street": "Hubala 10",
    "description": "<div><!--block-->The projects, you will work with, come from the most creative and innovative companies in the Silicon Valley, CA, USA and Western Europe. Take part in the things you have read about on the Internet and do the outstanding stuff here: in Poland. The real adventure is waiting for you.<br><br>About project:<br>Recognition of objects from drone photos, processing into 3D models and providing the user with platform to view and analize the results.<br><br></div><div><!--block-->Expected experience:</div><ul><li><!--block-->Very good spoken and written English</li><li><!--block-->Very good knowledge of Python or Node.JS</li><li><!--block-->Good understanding of Google Cloud, Circle CI and Kuberenetes for CI/CD</li><li><!--block-->Good knowledge of SQL and non-SQL databases</li><li><!--block-->Easy to work with</li><li><!--block-->Ambitious and excited to do new stuff</li></ul><div><!--block--><br>Appreciated knowledge:</div><ul><li><!--block-->Good practices in embedded software development</li><li><!--block-->Good understanding of microservices</li><li><!--block-->Experience in leading a small developers team</li></ul><div><!--block-->&nbsp;</div><div><!--block-->We offer:</div><ul><li><!--block-->International cutting edge projects – work with the technology leaders from CA, USA!</li><li><!--block-->Very flexible working hours, home-office during selected days possible</li><li><!--block-->Corporate-free environment, straightforward, simple, easy</li><li><!--block-->Long-term commitment from apptimia side</li></ul>",
    "title": "C Developer",
    "company": "Fine",
    "minSalary": 3000,
    "maxSalary": 5500,
    "companySize": "20 - 10 ",
    "website": "www.fine.com",
    "agreements": "Informujemy, że administratorem danych jest xD z siedzibą w ___________, ul. _______________ (dalej jako \"administrator\"). Masz prawo do żądania dostępu do swoich danych osobowych, ich sprostowania, usunięcia lub ograniczenia przetwarzania, prawo do wniesienia sprzeciwu wobec przetwarzania, a także prawo do przenoszenia danych oraz wniesienia skargi do organu nadzorczego. Dane osobowe przetwarzane będą w celu realizacji procesu rekrutacji. Podanie danych w zakresie wynikającym z ustawy z dnia 26 czerwca 1974 r. Kodeks pracy jest obowiązkowe. W pozostałym zakresie podanie danych jest dobrowolne. Odmowa podania danych obowiązkowych może skutkować brakiem możliwości przeprowadzenia procesu rekrutacji. Administrator przetwarza dane obowiązkowe na podstawie ciążącego na nim obowiązku prawnego, zaś w zakresie danych dodatkowych podstawą przetwarzania jest zgoda. Dane osobowe będą przetwarzane do czasu zakończenia postępowania rekrutacyjnego i przez okres możliwości dochodzenia ewentualnych roszczeń, a w przypadku wyrażenia zgody na udział w przyszłych postępowaniach rekrutacyjnych - do czasu wycofania tej zgody. Zgoda na przetwarzanie danych osobowych może zostać wycofana w dowolnym momencie. Odbiorcą danych jest serwis Just Join IT oraz inne podmioty, którym powierzyliśmy przetwarzanie danych w związku z rekrutacją."
  }
]

export const CV = [
  {
    "name": "Sylwester Kowalski",
    "email": "sylwesterkowalski@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": false,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a400.pdf"
  },
  {
    "name": "Kinga Sylwestrzak",
    "email": "kingasylwestrzak@gmail.com",
    "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet convallis massa. Curabitur efficitur pretium quam ut laoreet. Ut mollis ut velit quis semper. Nam placerat ullamcorper sapien vel laoreet. Vivamus lobortis elit id massa porta condimentum. Curabitur quis nulla eu purus egestas vestibulum. Vivamus semper arcu et blandit faucibus. Fusce sit amet massa cursus, tincidunt leo ut, molestie ligula. Proin sed nisi aliquet, facilisis velit at, scelerisque nunc. Integer sapien leo, porta non pellentesque id, aliquet in magna. Vestibulum venenatis porttitor enim a consequat. Maecenas sed malesuada dolor. Vivamus facilisis laoreet sapien, eu tincidunt dolor bibendum vitae. Etiam id sodales est. Phasellus consequat quam eu nunc tempor auctor.",
    "checkbox": true,
    "offerID": "5e9cfa222f9c0e0965b86034",
    "file": "9fce5a67910bd4d71cb75a401.pdf"
  }
]
