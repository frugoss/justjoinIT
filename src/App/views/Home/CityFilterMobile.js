import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Slide from '@material-ui/core/Slide';
import FilterButtonMobile from "./FilterButtonMobile";
import styles from "./home.module.scss";


const useStyles = makeStyles({
    root: {
        borderRadius: 20,
        textTransform: "none",
        letterSpacing: 0.5,
        fontSize: 12.6,
        color: '#777777',
        height: 35,
        background: "white",
        padding: '0 12px',
        borderColor: "rgb(228, 232, 240)",
        '&:hover': {
            borderColor: "rgb(186, 104, 200)",
            backgroundColor: "white"
        }
    },
    close: {
        position: "fixed",
        left:20,
        marginDown: 10
    }
});
const Transition = React.forwardRef(function Transition(props, ref) {
    return <Slide direction="down" ref={ref} {...props} />;
});
const CityFilterMobile = ({ filters, cities, updateFilters}) => {
    const buttonClass = useStyles()

    const text = "Location"
      const [open, setOpen] = React.useState(false);

      const handleClickOpen = () => {
          setOpen(true);
      };

      const handleClose = () => {
          setOpen(false);
      };
    const isActive = (link, filter) => {
        return filter === link
    };
const polishCity = cities.filter(city => city.to === filters.city)
return (

<>
    <Button className={polishCity[0] && isActive(polishCity[0].to, filters.city) && polishCity[0].to !== "all" ? `${styles.activeButton} ${buttonClass.root}` : buttonClass.root} variant="outlined" onClick={handleClickOpen}>
        {polishCity[0] && polishCity[0] && polishCity[0].name !== "All" ? polishCity[0].name : "Location" }
    </Button>
    <Dialog fullScreen open={open} onClose={handleClose} TransitionComponent={Transition}>
            <div onClick={handleClose}>
                <IconButton className={buttonClass.close} edge="start" color="inherit" onClick={handleClose} aria-label="close">
                    <CloseIcon />
                </IconButton>
        <h2 className={styles.textTitleMobile}>{text}</h2>
            </div>
        <Divider style={{marginTop:45}} />
            <div className={styles.buttonListMobile}>
        {cities && cities.map(({name, to}) => (

                <FilterButtonMobile filters={filters} name={name} to={to} key={to} setOpen={setOpen}
                              updateFilters={updateFilters} isActive={isActive(to, filters.city)}>
                </FilterButtonMobile>

            )
        )}
            </div>

    </Dialog>
</>
);
  }

export default CityFilterMobile;