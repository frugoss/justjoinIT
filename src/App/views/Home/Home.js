import React, {useEffect, useState} from 'react';
import Filters from './Filters';
import {withRouter} from 'react-router-dom';
import styles from './home.module.scss'

const Home = ({location, match, history, offers, fetching}) => {
    const [filters, setFilters] = useState({ city: match.params.city || "all",
        language: match.params.language || "all",
        experience: match.params.experience || "all",
        salarymin: match.params.salarymin || 0,
        salarymax: match.params.salarymax || 50,
    });

useEffect(() => {
    if (location.pathname === "/"){
        setFilters({city: "all", language: "all", experience: "all", salarymin: 0, salarymax: 50})
    }
}, [location.pathname])


    const updateFilter = (key,value) => {
        setFilters({...filters, [key]:value})
    };

    return (
        <div className={styles.filters}>
            <Filters offers={offers} fetching={fetching} updateFilters={updateFilter} setFilters={setFilters} filters={filters} history={history} match={match}/>
        </div>


    )
};


export default withRouter(Home)
