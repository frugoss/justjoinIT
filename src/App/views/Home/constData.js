import JS from "../../assets/images/JS.png";
import HTML from "../../assets/images/HTML.png";
import PHP from "../../assets/images/Php.png";
import Ruby from "../../assets/images/Ruby.png";
import Python from "../../assets/images/Python.png";
import Java from "../../assets/images/Java.png";
import Net from "../../assets/images/Net.png";
import Scala from "../../assets/images/Scala.png";
import C from "../../assets/images/C++.png";
import Mobile from "../../assets/images/Mobile.png";
import Testing from "../../assets/images/Testing.png";
import DevOps from "../../assets/images/DevOps.png";
import UX from "../../assets/images/UX:UI.png";
import PM from "../../assets/images/PM.png";
import Game from "../../assets/images/Game.png";
import SecurityIcon from "../../assets/images/Security.png";
import Data from "../../assets/images/Data.png";
import Go from "../../assets/images/Goo.png";
import SAP from "../../assets/images/SAP.png";
import Other from "../../assets/images/Other.png";

export const cities = [{
    name: "All",
    to: "all",
    coordinates: [51.92, 19.13],
}, {name: "Warszawa", to: "warszawa", coordinates: [52.229676, 21.012229]}, {
    name: "Wrocław",
    to: "wroclaw",
    coordinates: [51.10789, 17.03854]
}, {
    name: "Kraków",
    to: "krakow", coordinates: [50.06465, 19.94498]
}, {name: "Poznań", to: "poznan", coordinates: [52.40637, 16.92517]}, {
    name: "Gdańsk",
    to: "gdansk",
    coordinates: [54.35203, 18.64664]
}, {
    name: "Gdynia",
    to: "gdynia", coordinates: [54.51889, 18.53054]
}, {name: "Sopot", to: "sopot", coordinates: [54.44158, 18.56010]}, {
    name: "Szczecin",
    to: "szczecin",
    coordinates: [53.42894, 14.55302]
}];

export const language = [{
    name: "JS",
    to: "javascript",
    img: JS,
    color: "rgb(255, 199, 6)",
    background: "rgb(255, 199, 6), rgb(255, 175, 0)"
}, {
    name: "HTML",
    to: "html",
    img: HTML,
    color: "rgb(255, 125, 80)",
    background: "rgb(255, 125, 80), rgb(253, 93, 33)"
}, {
    name: "PHP",
    to: "php",
    img: PHP,
    color: "rgb(65, 173, 250)",
    background: "rgb(65, 173, 250), rgb(21, 124, 252)"
}, {name: "Ruby", to: "ruby", img: Ruby, color: "rgb(239, 83, 80)", background: "rgb(239, 83, 80), rgb(244, 67, 54)"}, {
    name: "Python",
    to: "python",
    img: Python,
    color: "rgb(31, 94, 182)",
    background: "rgb(31, 94, 182), rgb(31, 123, 196)"
}, {
    name: "Java",
    to: "java",
    img: Java,
    color: "rgb(250, 101, 107)",
    background: "rgb(250, 101, 107), rgb(249, 89, 122)"
}, {
    name: ".Net",
    to: "net",
    img: Net,
    color: "rgb(63, 197, 230)",
    background: "rgb(63, 197, 230), rgb(79, 164, 253)"
}, {
    name: "Scala",
    to: "scala",
    img: Scala,
    color: "rgb(248, 100, 104)",
    background: "rgb(248, 100, 104), rgb(241, 70, 76)"
}, {
    name: "C",
    to: "c",
    img: C,
    color: "rgb(47, 207, 187)",
    background: "rgb(47, 207, 187), rgb(55, 221, 156)"
}, {
    name: "Mobile",
    to: "mobile",
    img: Mobile,
    color: "rgb(224, 79, 134)",
    background: "rgb(224, 79, 134), rgb(186, 74, 141)"
}, {
    name: "Testing",
    to: "testing",
    img: Testing,
    color: "rgb(137, 219, 84)",
    background: "rgb(137, 219, 84), rgb(101, 175, 53)"
}, {
    name: "DevOps",
    to: "devops",
    img: DevOps,
    color: "rgb(82, 102, 225)",
    background: "rgb(82, 102, 225), rgb(129, 102, 224)"
}, {name: "UX/UI", to: "ux", img: UX, color: "rgb(255, 183, 77)", background: "rgb(255, 183, 77), rgb(255, 152, 0)"}, {
    name: "PM",
    to: "pm",
    img: PM,
    color: "rgb(128, 203, 196)",
    background: "rgb(128, 203, 196), rgb(77, 182, 172)"
}, {
    name: "Game",
    to: "game",
    img: Game,
    color: "rgb(255, 64, 129)",
    background: "rgb(255, 64, 129), rgb(236, 64, 122)"
}, {
    name: "Security",
    to: "security",
    img: SecurityIcon,
    color: "rgb(83, 109, 254)",
    background: "rgb(83, 109, 254), rgb(13, 71, 161)"
}, {
    name: "Data",
    to: "data",
    img: Data,
    color: "rgb(0, 150, 136)", background: "rgb(0, 150, 136), rgb(0, 121, 107)"
}, {name: "Go", to: "go", img: Go, color: "rgb(106, 215, 229)", background: "rgb(106, 215, 229), rgb(106, 139, 229)"}, {
    name: "SAP",
    to: "SAP",
    img: SAP,
    color: "rgb(2, 175, 235)",
    background: "rgb(2, 175, 235), rgb(27, 104, 194)"
}, {
    name: "Other",
    to: "Other",
    img: Other,
    color: "rgb(236, 76, 182)",
    background: "rgb(236, 76, 182), rgb(212, 75, 213)"
}];