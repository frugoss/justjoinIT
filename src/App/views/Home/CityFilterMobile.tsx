import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import Slide from '@material-ui/core/Slide';
import FilterButtonMobile from "App/views/Home/FilterButtonMobile";
import styles from "./home.module.scss";
import { TransitionProps } from '@material-ui/core/transitions';
import {filtersInterface, cityInterface} from "../../utils/const";


const useStyles = makeStyles(createStyles({
    root: {
        borderRadius: 20,
        textTransform: "none",
        letterSpacing: 0.5,
        fontSize: 12.6,
        color: '#777777',
        height: 35,
        background: "white",
        padding: '0 12px',
        borderColor: "rgb(228, 232, 240)",
        '&:hover': {
            borderColor: "rgb(186, 104, 200)",
            backgroundColor: "white"
        }
    },
    close: {
        position: "fixed",
        left:20,
        marginDown: 10
    },
    divider: {
        marginTop:45
    }
}));


const Transition = React.forwardRef(function Transition(
    props: TransitionProps & { children?: React.ReactElement },
    ref: React.Ref<unknown>,
)  {
    return <Slide direction="down" ref={ref} {...props} />;
});

type CityFilterMobile = {
    filters: filtersInterface,
    updateFilters: (key:string, value:string) => void,
    cities: cityInterface[]
}
const CityFilterMobile: React.FC<CityFilterMobile> = ({ filters, cities, updateFilters}) => {
    const classes = useStyles()

    const text = "Location"
      const [open, setOpen] = React.useState(false);

      const handleClickOpen = () => {
          setOpen(true);
      };

      const handleClose = () => {
          setOpen(false);
      };
    const isActive = (link:string, filter:string): boolean => {
        return filter === link
    };
const polishCity = cities.filter(city => city.to === filters.city)
return (
<>
    <Button className={polishCity[0] && isActive(polishCity[0].to, filters.city) && polishCity[0].to !== "all" ? `${styles.activeButton} ${classes.root}` : classes.root} variant="outlined" onClick={handleClickOpen}>
        {polishCity[0] && polishCity[0] && polishCity[0].name !== "All" ? polishCity[0].name : "Location" }
    </Button>
    <Dialog fullScreen open={open} onClose={handleClose} TransitionComponent={Transition}>
            <div onClick={handleClose}>
                <IconButton className={classes.close} edge="start" color="inherit" onClick={handleClose} aria-label="close">
                    <CloseIcon />
                </IconButton>
        <h2 className={styles.textTitleMobile}>{text}</h2>
            </div>
        <Divider className={classes.divider} />
            <div className={styles.buttonListMobile}>
        {cities && cities.map(({name, to}) => (

                <FilterButtonMobile filters={filters} name={name} to={to} key={to} setOpen={setOpen}
                              updateFilters={updateFilters} isActive={isActive(to, filters.city)}>
                </FilterButtonMobile>

            )
        )}
            </div>

    </Dialog>
</>
);
  }

export default CityFilterMobile;
