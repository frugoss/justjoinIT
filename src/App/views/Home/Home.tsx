import React, {useEffect, useState} from 'react';
import Filters from './Filters';
import {withRouter, RouteComponentProps} from 'react-router-dom';
import styles from './home.module.scss'
import {matchInterface, offerInterface} from "../../utils/const";



interface HomeProps extends RouteComponentProps<matchInterface>{
    offers: offerInterface[],
    fetching: boolean
}


const Home: React.FC<HomeProps> = ({location, match, offers, fetching, history}) => {
    const [filters, setFilters] = useState({ city: match.params.city || "all",
        language: match.params.language || "all",
        experience: match.params.experience || "all",
        salarymin: parseInt(match.params.salarymin, 10) || 0,
        salarymax: parseInt(match.params.salarymax, 10) || 50,
    });

useEffect(() => {
    if (location.pathname === "/"){
        setFilters({city: "all", language: "all", experience: "all", salarymin: 0, salarymax: 50})
    }
}, [location.pathname])


    const updateFilter = (key:string,value:string | number) => {
        setFilters({...filters, [key]:value})
    };

    return (
        <div className={styles.filters}>
            <Filters history={history} offers={offers} fetching={fetching} updateFilters={updateFilter} setFilters={setFilters} filters={filters}/>
        </div>


    )
};


export default withRouter(Home)
