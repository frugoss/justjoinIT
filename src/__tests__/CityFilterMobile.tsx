import React from 'react'
import {renderWithRouter} from "../App/utils/tests";
import CityFilterMobile from "../App/views/Home/CityFilterMobile";
import {cities} from "../App/views/Home/constData";



test('Snapshot',async() => {
    const {container} = renderWithRouter(<CityFilterMobile cities={cities} filters={{city:"all", language:"all", experience:"all", salarymax: 50, salarymin:1}} updateFilters={() => {}}/>)
    await expect(container).toMatchSnapshot()
})
