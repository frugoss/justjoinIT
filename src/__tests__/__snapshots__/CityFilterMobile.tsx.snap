// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-root-1"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Location
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
